scripts/scan-pdf-content.py

```diff
@@ -0,0 +1,300 @@
+    :return: None
+    """
+    while not scan_queue.empty():
+        i, pdf_summary = scan_queue.get()
+
+        if i % 10 == 0:
+            logging.info(f"Scanning file {i}")
+
+        pdf_content = fetch_s3_object_to_memory(pdf_summary)
+        code, message = scan_object(verapdf_url, pdf_content)
+        scan_result = ScanResult(
+            scanned_at=str(datetime.now()),
+            status_code=code,
+            message=message,
+            bucket=pdf_summary.bucket_name,
+            framework=framework,
+            key=pdf_summary.key,
+        )
+
+        write_result(report_name, scan_result)
+        scan_queue.task_done()
+
+
+def scan_all_pdfs(
+    bucket_name: str,
+    verapdf_url: str,
+    framework: str,
+    report_name: str,
+    index: Optional[int] = None
+) -> None:
+    """
+    Scan all pdfs in an S3 bucket against verapdf to see if any
+    contain JS, videos or any other unusual content types.
+
+    :param bucket_name: The S3 bucket to scan
+    :param verapdf_url: The base url to a verapdf-rest service
+    :param framework: The framework slug to scan, eg. g-cloud-12
+    :param report_name: The file to write scan results to
+    :param index: An optional index to resume from. If supplied, a list of PDFs to scan is read from
+                    '/tmp/pdf-scan-queue.json' and the scan is started from position `index`.
+    :return: None
+    """
+    if index is not None:
+        # Resuming from a previous run. Load files to scan from disk and set up index
+        logging.info("Resuming from previous run")
+        pdfs_to_scan = load_pdfs_to_scan_from_file()
+
+        logging.info(f"Resuming scan from position: {index}")
+        pdfs_to_scan = pdfs_to_scan[index:]
+
+    else:
+        # Starting new scan. Get files to scan from S3
+        pdfs_to_scan = list_pdfs_in_bucket(bucket_name, framework)
+
+    total_to_scan = len(pdfs_to_scan)
+
+    save_all_pdf_names_to_scan(pdfs_to_scan)
+    logging.info(f"Scanning {total_to_scan} files")
+
+    # Put all tasks into a queue
+    scan_queue = queue.Queue()
+    for i, pdf in enumerate(pdfs_to_scan):
+        scan_queue.put((i, pdf))
+
+    # Start threads
+    for _ in range(3):
+        threading.Thread(
+            target=scan_pdf,
+            args=(scan_queue, verapdf_url, report_name)
+        ).start()
+
+    # Wait for threads to finish
+    scan_queue.join()
+
+    logging.info("Finished scanning")
+    logging.info("Removing temporary files")
+    os.remove("/tmp/pdf-scan-queue.json")
+
+
+def set_up_report_file(filename: str) -> None:
+    """
+    Create an empty CSV with headers to store results
+
+    :param filename: The name of the output file to create
+    :returns: None
+    """
+    with open(filename, "w") as f:
+        report_writer = csv.writer(f, delimiter=",")
+        report_writer.writerow(
+            ["scanned_at", "bucket", "framework", "key", "status_code", "message"]
+        )
```

Comment:
It might be good to move this further up the file to be close to `write_result`, so you can read them together and more quickly understand how they work together!